### PR TITLE
feat(redis): update to v9 and add context

### LINKED ISSUE #132: redis: update to v9
see https://pkg.go.dev/github.com/redis/go-redis/v9

### PR DESCRIPTION
**Pull request by glimchb**
following https://github.com/philippgille/gokv/commit/9c5b7d72f129ab5b7df52ef80b7ae280ace2db80
Resolves #132 

Signed-off-by: Boris Glimcher <Boris.Glimcher@emc.com>

### CODE REVIEW

**Comment by glimchb** [comment]
@philippgille please review

File: redis/go.mod [hunk 1]
```diff
@@ -7,9 +7,12 @@ require (
 	github.com/philippgille/gokv/encoding v0.6.0
 	github.com/philippgille/gokv/test v0.6.0
 	github.com/philippgille/gokv/util v0.6.0
+	github.com/redis/go-redis/v9 v9.2.1
```
Comment (philippgille) [inline redis/go.mod:10]: With this update, the old v6 dependency shouldn't be required anymore, right? Did you run `go mod tidy`?

File: redis/redis.go [hunk 2]
```diff
@@ -30,7 +33,10 @@ func (c Client) Set(k string, v interface{}) error {
 		return err
 	}
 
-	err = c.c.Set(k, string(data), 0).Err()
+	tctx, cancel := context.WithTimeout(context.Background(), 2*time.Second)
```
Comment (philippgille) [inline redis/redis.go:36]: I would prefer one of the following:
- Not break old behavior, use context without timeout
- Extend the config with a configurable timeout. Some gokv packages already do that, like `etcd` one [here](https://github.com/philippgille/gokv/blob/93d491e1a0c9ba21a1a88e7e5888da159671a122/etcd/etcd.go#L101)

And in the future I think we should change `gokv` signatures to one with a context as first parameter, following best practices and giving users full control of per-call timeouts.

**Review by philippgille — COMMENTED** [review]
Thanks for the PR!

So far I was thinking to just update to latest minor+patch versions for the next `gokv` release, and then bump major versions in the following one. This would keep breaking changes low and people could update from the really old v0.6.0 to a fresh v0.7.0 without worrying too much but getting important security updates.

But OTOH maybe we don't need to follow the same strategy for all packages, for example if there are no breaking changes. Have you checked if in the Redis library v6 to v9 there's anything that could affect gokv users? We're only using a very limited subset of the Redis functionalities, so I could imagine we're not affected, but better to double check.

WDYT?

File: redis/redis.go (hunk 2 above)
Comment (glimchb) [reply redis/redis.go:36]: thanks, sure I can add timeout config!

I was following https://github.com/philippgille/gokv/blob/99127ded57972e470ac9d4a88008c0c9f4d8b8a6/dynamodb/dynamodb.go#L254 and https://github.com/philippgille/gokv/blob/99127ded57972e470ac9d4a88008c0c9f4d8b8a6/datastore/datastore.go#L45 that have hard-coded timeout, but I can follow https://github.com/philippgille/gokv/blob/93d491e1a0c9ba21a1a88e7e5888da159671a122/etcd/etcd.go#L101 and also update the other two to add timeout configuration

Yes, context as first parameter is much needed at this point!

File: redis/go.mod (hunk 1 above)
Comment (glimchb) [reply redis/go.mod:10]: I thought I did. Double-checking now...

**Comment by glimchb** [comment]
> Thanks for the PR!
> 
> So far I was thinking to just update to latest minor+patch versions for the next `gokv` release, and then bump major versions in the following one. This would keep breaking changes low and people could update from the really old v0.6.0 to a fresh v0.7.0 without worrying too much but getting important security updates.
> 
> But OTOH maybe we don't need to follow the same strategy for all packages, for example if there are no breaking changes. Have you checked if in the Redis library v6 to v9 there's anything that could affect gokv users? We're only using a very limited subset of the Redis functionalities, so I could imagine we're not affected, but better to double check.
> 
> WDYT?

Excellent comments!

The main reason to update to v9 was due to otel in #131

Sorry, I haven't checked the Redis library v6 to v9 for breaking changes

File: redis/redis.go (hunk 2 above)
Comment (philippgille) [reply redis/redis.go]: Yes indeed, not all store implementations have a configurable timeout yet. Thanks for the update!

**Comment by glimchb** [comment]
> Can you update the branch with the latest master changes? If CI is 🟢 (at least for the Redis store) then it's good to go 🚀

CI passed @philippgille !

**Review by philippgille — APPROVED** [review]
Thanks again :+1:

**Pull request merged**